src/lavinmq/http/controller/connections.cr

```diff
@@ -48,6 +48,27 @@ module LavinMQ
             page(context, connection.channels.each_value)
           end
         end
+
+        get "/api/connections/username/:username" do |context, params|
+          connections = get_connections_by_username(context, params)
+          page(context, connections.each)
+        end
+
+        delete "/api/connections/username/:username" do |context, params|
+          connections = get_connections_by_username(context, params)
+          reason = context.request.headers["X-Reason"]? || "Closed via management plugin"
+          connections.each do |c|
+            c.close(reason)
+          end
+          context.response.status_code = 204
+          context
+        end
+      end
+
+      private def get_connections_by_username(context, params)
```

Comment:
Method says `by_username` but you are giving it `params` So I think you should extract username in the caller method and give this method a "username" not params. 
Otherwise the method should be called `get_connections_by_params` but that is really difficult to understand.